### PR TITLE
Added support for CCEE infra

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added support for CCEE infra, so that it can scrape a cluster on OpenStack.

image

**Related issue(s)**

- https://github.com/kyma-project/control-plane/issues/3179

### CODE REVIEW

File: components/kyma-metrics-collector/pkg/process/public_cloud_info.go [hunk 1]
```diff
@@ -85,11 +92,20 @@ func LoadPublicCloudSpecs(cfg *env.Config) (*Providers, error) {
 	if err = json.Unmarshal(gcpMachinesData, gcpMachines); err != nil {
 		return nil, errors.Wrapf(err, "failed to unmarshal GCP machines data")
 	}
+	openStackMachinesData, err := machineInfo[OpenStack].MarshalJSON()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to marshal OpenStack info")
+	}
+	openStackMachines := &OpenStackMachines{}
+	if err = json.Unmarshal(openStackMachinesData, openStackMachines); err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal GCP machines data")
```
Comment (marcobebway) [inline components/kyma-metrics-collector/pkg/process/public_cloud_info.go:101]:
```suggestion
		return nil, errors.Wrapf(err, "failed to unmarshal OpenStack machines data")
```

File: components/kyma-metrics-collector/pkg/process/event_stream_test.go [hunk 2]
```diff
@@ -121,12 +121,43 @@ func TestParse(t *testing.T) {
...
+			name: "with OpenStack, 2 vm types, 3 pvcs(5,10 and 20Gi), and 2 svcs(1 clusterIP and 1 LoadBalancer)",
+			input: Input{
+				shoot:    kmctesting.GetShoot("testShoot", kmctesting.WithOpenStackProviderAndMachineGC12M48),
+				nodeList: kmctesting.Get2NodesOpenStack(),
+				pvcList:  kmctesting.Get3PVCs(),
+				svcList:  kmctesting.Get2SvcsOfDiffTypes(),
+			},
+			providers: *providers,
+			expectedMetrics: edp.ConsumptionMetrics{
+				//ResourceGroups: nil,
```
Comment (marcobebway) [inline components/kyma-metrics-collector/pkg/process/event_stream_test.go:134]: Please delete if not needed.

File: components/kyma-metrics-collector/pkg/testing/fixtures/static_providers.json [hunk 3]
```diff
@@ -1,213 +1,329 @@
 {
   "azure": {
     "standard_a1_v2": {
-        "cpu_cores": 1,
-        "memory": 2,
-        "storage": 10,
-        "max_nics": 1
+      "cpu_cores": 1,
```
Comment (marcobebway) [inline components/kyma-metrics-collector/pkg/testing/fixtures/static_providers.json:4]: Is there a document describing how to get the values in this file?

File: resources/kcp/charts/kyma-metrics-collector/templates/public-cloud-info-config-map.yaml [hunk 4]
```diff
@@ -357,7 +357,7 @@ data:
          "memory": 320
        }
      },
-      "sap-converged-cloud": {
+      "openstack": {
```
Comment (marcobebway) [inline resources/kcp/charts/kyma-metrics-collector/templates/public-cloud-info-config-map.yaml:360]: Why the `sap-converged-cloud` value changed?

Comment (mfaizanse) [reply resources/kcp/charts/kyma-metrics-collector/templates/public-cloud-info-config-map.yaml:360]: It was added in [PR](https://github.com/kyma-project/control-plane/pull/3247). But it is incorrect.

File: components/kyma-metrics-collector/pkg/testing/fixtures/static_providers.json (hunk 3 above)
Comment (mfaizanse) [reply components/kyma-metrics-collector/pkg/testing/fixtures/static_providers.json:4]: The values are copied it from helm chart https://github.com/kyma-project/control-plane/blob/main/resources/kcp/charts/kyma-metrics-collector/templates/public-cloud-info-config-map.yaml.

The machine types are manually updated as needed. For example: https://github.com/orgs/kyma-project/projects/17/views/7?pane=issue&itemId=51280791

File: components/kyma-metrics-collector/pkg/process/public_cloud_info.go [hunk 5]
```diff
@@ -21,6 +22,8 @@ type AWSMachines map[string]Feature
 
 type GCPMachines map[string]Feature
 
+type OpenStackMachines map[string]Feature
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply components/kyma-metrics-collector/pkg/process/public_cloud_info.go:25]: Not sure, why the naming was used in the first place. But I just followed same pattern to be consistent. Also, we also have structs for `MachineInfo` and `Feature`. I guess, the purpose was to parse the json data.

**Review by marcobebway — APPROVED** [review]

**Pull request merged**